File: src/main/java/org/voegtle/weatherwidget/data/WeatherJSONObject.java
package org.voegtle.weatherwidget.data;

import org.json.JSONException;
import org.json.JSONObject;

public class WeatherJSONObject extends JSONObject {

  @Override
  public JSONObject put(String key, Object value) throws JSONException {
    if (value == null) {
      super.put(key, "");
    } else {
      return super.put(key, value);
    }
    return this;
  }

}

File: src/main/java/org/voegtle/weatherwidget/data/WeatherJSONObject.kt
package org.voegtle.weatherwidget.data

import org.json.JSONException
import org.json.JSONObject

class WeatherJSONObject : JSONObject() {

  @Throws(JSONException::class)
  override fun put(key: String, value: Any?): JSONObject {
    return super.put(key, value?: "")
  }

}
